src/hotspot/share/opto/macro.cpp

```diff
@@ -2433,6 +2433,9 @@ bool PhaseMacroExpand::expand_macro_nodes() {
   // Last attempt to eliminate macro nodes.
   eliminate_macro_nodes();
   if (C->failing())  return true;
+  if (StressMacroExpansion) {
+    C->shuffle_macro_nodes();
+  }
```

Comment:
I suggest to do this already before `eliminate_macro_nodes()` to also get a different macro node elimination order if multiple nodes could be removed.